pkg/windows/windows.go

```diff
@@ -417,44 +413,29 @@ func (vm *windows) EnsureRequiredServicesStopped() error {
 	return nil
 }
 
-// ensureServicesAreRemoved ensures that all services installed by WMCO are removed from the instance
-func (vm *windows) ensureServicesAreRemoved() error {
-	// TODO: In this transitional period of migrating services to WICD's responsibility, WICD must be stopped first to
-	//       ensure it does not start any of these stopped services. This code should be removed as part of
-	//       https://issues.redhat.com/browse/WINC-733
-	svc := &service{name: wicdServiceName}
-	if err := vm.ensureServiceNotRunning(svc); err != nil {
-		return errors.Wrapf(err, "could not stop service %s", wicdServiceName)
+func (vm *windows) RunWICDCleanup(apiServerURL string, preserveNode bool) error {
+	wicdCleanupCmd := fmt.Sprintf("%s cleanup --api-server %s --sa-ca %s --sa-token %s",
+		wicdPath, apiServerURL, wicdCAFile, wicdTokenFile)
+	if preserveNode {
+		wicdCleanupCmd = wicdCleanupCmd + " --preserveNode"
 	}
-	for _, svcName := range append(RequiredServices, AzureCloudNodeManagerServiceName) {
-		svc := &service{name: svcName}
-
-		// If the service is not installed, do nothing
-		exists, err := vm.serviceExists(svc.name)
-		if err != nil {
-			return errors.Wrapf(err, "unable to check if %s service exists", svc.name)
-		}
-		if !exists {
-			continue
-		}
-
-		// Make sure the service is stopped before we attempt to delete it
-		if err := vm.ensureServiceNotRunning(svc); err != nil {
-			return errors.Wrapf(err, "could not stop service %s", svc.name)
-		}
-		if err := vm.deleteService(svc); err != nil {
-			return errors.Wrapf(err, "could not delete service %s", svcName)
-		}
-		vm.log.Info("deconfigured", "service", svc.name)
+	if out, err := vm.Run(wicdCleanupCmd, true); err != nil {
+		vm.log.Info("failed to cleanup node", "command", wicdCleanupCmd, "output", out)
+		return err
 	}
 	return nil
 }
 
-func (vm *windows) Deconfigure() error {
+func (vm *windows) Deconfigure(preserveNode bool, apiServerURL string) error {
 	vm.log.Info("deconfiguring")
-	if err := vm.ensureServicesAreRemoved(); err != nil {
-		return errors.Wrap(err, "unable to remove Windows services")
+
+	if err := vm.deconfigureWICD(); err != nil {
+		return err
+	}
+	if err := vm.RunWICDCleanup(apiServerURL, preserveNode); err != nil {
+		return errors.Wrap(err, "Unable to cleanup the Windows instance")
 	}
+	time.Sleep(time.Minute)
```

Comment:
Is this for debugging purposes or actually required?

Reply:
debugging only